### PR TITLE
Feature/ddbteam 377 expand stats logging

### PR DESCRIPTION
**Pull request by tuj**
* Adds matches array to elasticsearch stats logging.

### CODE REVIEW

File: src/Api/DataProvider/AbstractElasticSearchDataProvider.php [hunk 1]
```diff
@@ -244,15 +246,48 @@ protected function logStatistics(string $type, array $identifiers, array $result
...
+        $imageUrls = is_array($results) ? $this->getImageUrls($results) : [];
+
         $this->statsLoggingService->info('Cover request/response', [
             'service' => $className,
             'clientID' => 'REST_API',
             'remoteIP' => $request->getClientIp(),
             'isType' => $type,
             'isIdentifiers' => $identifiers,
-            'fileNames' => $this->getImageUrls($results),
+            'fileNames' => !empty($imageUrls) ? array_values($imageUrls) : null,
```
Comment (cableman) [inline src/Api/DataProvider/AbstractElasticSearchDataProvider.php]: Why not default to [] and not NULL then the data structure is the same.

File: src/Service/MoreInfoService/AbstractMoreInfoService.php [hunk 2]
```diff
@@ -262,7 +264,8 @@ public function moreInfo($body): MoreInfoResponse
 
         $results = [];
         if (false === $response) {
-            $this->logger->error('Curl ES query error: '.curl_error($ch));
+            $error = curl_error($ch);
+            $this->logger->error('Curl ES query error: '.$error);
```
Comment (cableman) [inline src/Service/MoreInfoService/AbstractMoreInfoService.php]: Why the extra variable?

File: src/Service/MoreInfoService/AbstractMoreInfoService.php [hunk 3]
```diff
@@ -272,12 +275,14 @@ public function moreInfo($body): MoreInfoResponse
...
 
         $time = microtime(true);
+        $imageUrls = is_array($results) ? $this->getImageUrls($results) : [];
         $this->statsLoggingService->info('Cover request/response', [
             'service' => 'MoreInfoService',
             'clientID' => $body->authentication->authenticationGroup,
             'remoteIP' => $this->requestStack->getCurrentRequest()->getClientIp(),
             'searchParameters' => $searchParameters,
-            'fileNames' => $this->getImageUrls($results),
+            'fileNames' => !empty($imageUrls) ? array_values($imageUrls) : null,
```
Comment (cableman) [inline src/Service/MoreInfoService/AbstractMoreInfoService.php]: Default to [] to keep same type.

File: src/Api/DataProvider/AbstractElasticSearchDataProvider.php (hunk 1 above)
Comment (tuj) [reply src/Api/DataProvider/AbstractElasticSearchDataProvider.php]: $this->getImage() returned null or array with entries before. 
So to not change the logging structure of "fileNames", I chose this solution, after I changed the return type of getImageUrls() from mixed to array.

File: src/Service/MoreInfoService/AbstractMoreInfoService.php (hunk 3 above)
Comment (tuj) [reply src/Service/MoreInfoService/AbstractMoreInfoService.php]: same as above comment

File: src/Api/DataProvider/AbstractElasticSearchDataProvider.php (hunk 1 above)
Comment (cableman) [reply src/Api/DataProvider/AbstractElasticSearchDataProvider.php]: We are changing the data structure... I think ES would be happier if the data always was an array.

File: src/Service/MoreInfoService/AbstractMoreInfoService.php (hunk 3 above)
Comment (cableman) [reply src/Service/MoreInfoService/AbstractMoreInfoService.php]: same as above comment :-)

**Pull request merged**